### ISSUE TITLE
Pressure Vessel fails with execvp env: No such file or directory if env is not in PATH

### ISSUE THREAD

**Issue by mmoore7** [ValveSoftware/steam-runtime#337]
OS: Arch Linux
KERNEL: 5.9.14-arch1-1
CPU: Intel Core i7-8700K @ 3.70GHz
GPU: NVIDIA GeForce RTX 2080 Ti
GPU DRIVER: NVIDIA 455.45.07-1
RAM: 32 GB

* Steam Runtime Version: 0.20201203.1
* Distribution (e.g. Ubuntu 18.04): Arch 5.9.14
* Link to your full system information (Help -> System Information) in a [Gist](https://gist.github.com/): [Full System Information](https://gist.github.com/mmoore7/cf059ea0e8ef93140e1c287c49bb1de7)
  
* Have you checked for system updates?: Yes
* Are you using the Steam Linux Runtime compatibility tool?: Yes

#### Please describe your issue in as much detail as possible:

This was originally posted in the MHW issue tracker here: [MHW 582010](https://github.com/ValveSoftware/Proton/issues/175#issuecomment-748434327). Basically all my games no longer launch on Proton 5.13-4. Two games I've tried is **Monster Hunter World** and **Middle Earth Shadow of War**. They continue to work on older versions of proton as well as older versions of GE.

I've tried verifying game files and reinstalling the game, which didn't work. I noticed a sym link error in steam terminal output, but the link path in the error changes every time I attempt to run, so creating the file doesn't fix it either. 

This affects both Proton-5.21-GE-1 and Proton 5.13-4

This is the output from terminal:
[Terminal Output](https://gist.github.com/mmoore7/ecc02afbf9a3332afa065a1f30505dad)

Here is the pressure-vessel log: [pressure-vessel.log](https://gist.github.com/mmoore7/5e2708d98ca001ba52d960560dd59809)

Note: I do have most if not all my games installed on a 2TB SSD drive which is separate from my home directory drive.

#### Steps for reproducing this issue:

1. Launch Steam
2. Launch game
3. Nothing happens

**Comment by kisak-valve** [comment]
Hello @mmoore7, can check if opting into the client_beta branch of Steam Linux Runtime - Soldier has an effect? You can find it by searching for `runtime` in the Library section of Steam, right click on it -> Properties -> Betas -> client_beta

* pressure-vessel assumes that the `PATH` from outside the container is suitable to run `env` inside the container. Workaround: add `/usr/bin` or `/bin` to the `PATH`. (from https://github.com/ValveSoftware/steam-runtime/issues/230#issuecomment-733769126)

May be already fixed by https://gitlab.steamos.cloud/steamrt/steam-runtime-tools/-/merge_requests/188.

**Comment by mmoore7** [comment]
It looks like that worked. I'm able to launch Monster Hunter World on 5.13-4 now after opting into the client_beta.

**Comment by smcv** [comment]
Did Middle Earth Shadow of War have the same symptom mentioning `execvp env: No such file or directory`, and does it work now?

If yes then I think we can close this.

**Comment by mmoore7** [comment]
At the time of my last post, both Monster Hunter World and Shadow of War worked after opting into the client_beta. However, now neither work on 5.13-4, and it looks like the same pressure vessel error. I've updated my system at least 1-2 times since.

Here is my system info: [SystemInfo](https://gist.github.com/mmoore7/dbb580956100b2096ab578a28514f0e4)

Here's terminal output from attempting to launch Shadow of War: [Terminal Output](https://gist.github.com/mmoore7/2df4a62135842caae471ed024a13a8a9)

**Comment by smcv** [comment]
> Here's terminal output from attempting to launch Shadow of War

That's weird, `bwrap: execvp env: No such file or directory` has come back.

What's in `SteamLinuxRuntime_soldier/VERSIONS.txt`?

If you run Steam with `PATH="$PATH:/usr/bin" steam` as a workaround, does that succeed?

**Comment by smcv** [comment]
Ah, !188 didn't make it into a release yet, so it's expected that this is not fixed yet.

This might be fixed by the *next* Steam Linux Runtime - Soldier beta release.

**Comment by smcv** [comment]
@kisak-valve, please could you retitle this to mention `execvp env: No such file or directory` in the title?

**Comment by mmoore7** [comment]
Using `PATH="$PATH:/usr/bin" steam` did not work. Here's `SteamLinuxRuntime_soldier/VERSIONS.txt`:

```
#Name   Version         Runtime Runtime_Version Comment
SteamLinuxRuntime       v0.20201124.0-11-g197f448                       # Entry point scripts, etc.
pressure-vessel 0.20201203.0+srt1       scout   0.20201214.0    # pressure-vessel-bin.tar.gz
soldier 0.20201210.0    soldier 0.20201210.0    # com.valvesoftware.SteamRuntime.Platform-amd64,i386-soldier-runtime.tar.gz
```

**Comment by smcv** [comment]
I wonder whether this is another problem with the runtime not being fully unpacked - which isn't meant to be possible, but continues to happen intermittently.

If you rename `soldier_0.20201210.0` to `broken_0.20201210.0` and try to launch again (which will unpack a new copy of the runtime), does that work any better?

**Comment by smcv** [comment]
Alternatively, the new beta (mentioning pressure-vessel 0.20210105.0 in VERSIONS.txt) *might* fix this. It's hard to say whether it does or not when we don't understand the root cause.

**Comment by smcv** [comment]
I think we might have found why this didn't work. If that's correct, then this should *really* be fixed in beta 0.20210114.2.

**Comment by kisak-valve** [comment]
Closing pending feedback.
